sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/user_credential_async.py

```diff
@@ -33,14 +35,29 @@ class CommunicationTokenCredential(object):
     _ON_DEMAND_REFRESHING_INTERVAL_MINUTES = 2
     _DEFAULT_AUTOREFRESH_INTERVAL_MINUTES = 10
 
-    def __init__(self, token: str, **kwargs: Any):
-        if not isinstance(token, str):
-            raise TypeError("Token must be a string.")
-        self._token = create_access_token(token)
-        self._token_refresher = kwargs.pop("token_refresher", None)
-        self._proactive_refresh = kwargs.pop("proactive_refresh", False)
-        if self._proactive_refresh and self._token_refresher is None:
-            raise ValueError("When 'proactive_refresh' is True, 'token_refresher' must not be None.")
+    def __init__(self, token: str = None, **kwargs: Any):
+        self._resource_endpoint = kwargs.pop("resource_endpoint", None)
+        self._token_credential = kwargs.pop("token_credential", None)
+        self._scopes = kwargs.pop("scopes", None)
+
+        if self._resource_endpoint and self._token_credential and self._scopes:
+            self._token_exchange_client = AsyncTokenExchangeClient(self._resource_endpoint, self._token_credential, self._scopes)
+            self._token_refresher = lambda: self._token_exchange_client.exchange_entra_token()
+            self._proactive_refresh = kwargs.pop("proactive_refresh", False)
+
+            async def _initialize_token():
+                self._token = await self._token_exchange_client.exchange_entra_token()
+
+            asyncio.get_event_loop().run_until_complete(_initialize_token())
```

Comment:
Why do we need to wait for completion? Isn't it enough if the next get_token calls waits for completion?

Reply:
Because _token needs to be initialized before the first call to get_token